lib/response.ex

```diff
@@ -12,16 +18,47 @@ defmodule HTTPServer.Response do
 
   @carriage_return "\r\n"
 
-  def send_resp(status_code, body, headers) do
-    %__MODULE__{
-      status_code: status_code,
-      status_message: status_message(status_code),
-      resource: "HTTP/1.1",
-      headers: Headers.collect_headers(headers),
-      body: body
-    }
+  def send_resp(resp) do
+    %{resp | headers: Headers.collect_headers(resp.headers)}
+  end
+
+  def build(req = %Request{method: "HEAD"}, status_code, res_body, media_type) do
+    resp =
+      %__MODULE__{}
+      |> resource()
+      |> status(status_code)
+      |> body(res_body)
+      |> headers(req, media_type)
+
+    send_resp(%{resp | body: ""})
+  end
```

Comment:
```suggestion
  def build(req = %Request{method: "HEAD"}, status_code, res_body, media_type) do
    resp = build(%Request{ req | method "GET}, status_code, res_body, media_type)
    send_resp(%{resp | body: ""})
  end
```

Reply:
a HEAD `build(req = %Request{method: "HEAD"}, status_code, res_body, media_type)`
is just a GET `resp = build(%Request{ req | method "GET}, status_code, res_body, media_type)`
with the body stripped away `send_resp(%{resp | body: ""})`

Reply:
🤯

Reply:
```suggestion
  def build(req = %Request{method: "HEAD"}, status_code, res_body, media_type) do
    resp = build(%Request{req | method: "GET"}, status_code, res_body, media_type)
    %{resp | body: ""}
  end
```
Since `send_resp` is already called in `build/4` it was throwing errors when I tried to call it again, but returning `%{resp | body: ""}` achieves the desired result!